api/src/main/java/com/datastrato/gravitino/tag/SupportsTags.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.tag;
+
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.exceptions.NoSuchTagException;
+
+/**
+ * Interface for supporting getting or associate tags to entities. This interface will be mixed with
+ * metadata objects to provide tag operations.
+ */
+@Evolving
+public interface SupportsTags {
+
+  /**
+   * List all the tag names for the specific entity.
+   *
+   * @return The list of tag names.
+   */
+  String[] listTags();
+
+  /**
+   * List all the tags with details for the specific entity.
+   *
+   * @return The list of tags.
+   */
+  Tag[] listTagsInfo();
+
+  /**
+   * Get a tag by its name for the specific entity.
+   *
+   * @param name The name of the tag.
+   * @return The tag.
+   */
+  Tag getTag(String name) throws NoSuchTagException;
+
+  /**
+   * Associate tags to the specific entity. The tagsToAdd will be added to the entity, and the
+   * tagsToRemove will be removed from the entity.
+   *
+   * @param tagsToAdd The arrays of tag name to be added to the entity.
+   * @param tagsToRemove The array of tag name to be removed from the entity.
+   * @return The array of tag names that are associated with the entity.
+   */
+  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove);
```

Comment:
for this method, the requestor need to know which tags are new, which are existed but to be removed, this might be a little complicated for the requestor; how about change it to "applyTags(String[] tags)", so that each time the requestor pass in a full set of the tags, no need to distinguish which are new and which are to be remove;

Reply:
I see, the backend will compare and recognize that, which is good for the client.

Reply:
If a tag appears in two arrays at the same time, what will be the result?

Reply:
It will be removed. Same as if we both enable `setProperty` and `removeProperty` on the same key for alterTable.